x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -417,8 +417,9 @@ else if (plan instanceof OrderBy) {
             for (NamedExpression ne : projections) {
                 if (ne instanceof UnresolvedStar) {
                     List<NamedExpression> expanded = expandStar((UnresolvedStar) ne, output);
+
                     // the field exists, but cannot be expanded (no sub-fields)
-                    if (expanded.isEmpty()) {
+                    if (expanded == null) {
```

Comment:
why not keep the empty list vs a null?

Reply:
That's more or less what caused the bug. Previously, expansion errors resulted in an empty list but that's ambiguous if the star is expanded on an index without columns. In this case, it returns an empty list meaning "star has been expanded to no fields".

To distinguish between expansion errors and empty expansions I now had to encode the error case with a `null` return value.

Reply:
You could return `singletonList(ne)` instead of passing null and treat it separately.

Reply:
got it, makes sense 👍